### DISCUSSION TITLE
IllegalStateException 발생 시 예외 코드와 로그 레벨은 어떻게 설정할까요?

### DISCUSSION THREAD

**Discussion by Kim0914** [woowacourse-teams/2023-pium#276] [뜨거운 토론]
친애하는 백엔드 여러분
* `IllegalStateException` 발생 시 **404 or 500** ?!
* log 레벨 `warn` or `error` 

우선 현재 ISE를 직접 throw하는 부분은 최근 물주기 정보는 하나만 존재해야 하는데 하나가 아닐때 입니다.
직접 핸들링 하는 예외를 500으로 던지는게 맞을까? 라는 생각이 들었습니다.

로그 레벨은 예상치 못한 동작이기 때문에 슬랙에 알림이 오는게 맞는 것 같아서 로그 레벨은 error로 설정하려고 합니다.

상태코드는 404, 500 중에 고민이 되는데요 !

404를 던지는 경우에는 정말 치명적인 예외가 아니기 때문에 서버 내부 상태를 숨긴다는 의도를 담을 수 있고
500을 던지는 경우에는 말 그대로 서버 내부에 문제가 생겼다는 의도를 담을 수 있다고 생각합니다.

요약하면 다음과 같습니다.

1. 로그 레벨을 error로 설정할까요? warn으로 설정할까요?
2. 상태 코드를 404를 던질까요? 500을 던질까요?

**Comment by Choi-JJunho** [comment]
로깅은 Error에 한표입니다!

ISE는 Db에 값이 잘못 들어간경우가 대부분일텐데 비즈니스 흐름으로 제어할 수 없는 부분이라 생각하여 error로 로깅하는게 좋다고 생각합니다

그렇지만 클라이언트에게 이러한 정보를 알려줄 이유는 없다고생각하여 반환값은 404여도 될거같네요

**Comment by rawfishthelgh** [comment]
1번은 error 2번은 404라고 판단이 듭니다.
500은 서버적으로 잘못된 로직이 작성되었을 경우나 이상이 발생할 경우 던져야 하는 오류인데
위와 같은 경우는 있어야 할 값이 없는 경우. 따라서 서버 로직과는 관련이 없는 것 같아요!
